src/main/scala/bi/deep/DruidDataReader.scala

```diff
@@ -21,22 +21,22 @@ case class DruidDataReader(filePath: String, schema: StructType, config: Config)
   }
 
   private lazy val rowConverter: DruidRowConverter = withSegment(filePath, config) { file =>
-    val qi = indexIO.loadIndex(new File(file.getAbsolutePath))
-    val qiia = new QueryableIndexIndexableAdapter(qi)
-    val qisa = new QueryableIndexStorageAdapter(qi)
-    val segmentSchema = DruidSchemaReader.readSparkSchema(qisa)
-    val rowDimensionsNames = segmentSchema.structTypeDimensions.fieldNames
-    val rowMetricsNames = segmentSchema.structTypeMetrics.fieldNames
-
-    val filteredTargetDimensions: Array[(StructField, Int)] = schema.fields
+    val queryableIndex = indexIO.loadIndex(new File(file.getAbsolutePath))
+    val queryableIndexIndexableAdapter = new QueryableIndexIndexableAdapter(queryableIndex)
+    val queryableIndexStorageAdapter = new QueryableIndexStorageAdapter(queryableIndex)
+    val druidSegmentSchema = DruidSchemaReader.readSparkSchema(queryableIndexStorageAdapter)
+    val druidDimensions = druidSegmentSchema.structTypeDimensions.fieldNames
+    val druidMetrics = druidSegmentSchema.structTypeMetrics.fieldNames
```

Comment:
One thing, please check if it is actual a druidSchema or sparkSchema, ok? Just by the types, but I guess now it is correct